openeogeotrellis/deploy/batch_job.py

```diff
@@ -211,6 +270,238 @@ def epsg_code(gps_crs) -> Optional[int]:
+        logger.warning(
+            "Could not get projection extension metadata, "
+            + f"gdal.Info failed for following asset: {asset_uri}. File is probably not a raster."
+            + f"Exception from GDAL: {exc}"
+        )
+        return None
+
+
+def parse_projection_extension_metadata(gdal_info: dict) -> dict:
+    """Parse the JSON output from gdal.Info.
+
+    :param gdal_info: Dictionary that contains the output from `gdal.Info()`.
+    :return: Dictionary containing the info for the STAC extension for projections.
+    """
+
+    # If there are subdatasets then the final answer comes from the subdatasets.
+    # Otherwise we get it from the file directly.
+    proj_info = _process_gdalinfo_for_netcdf_subdatasets(gdal_info)
+    if proj_info:
+        return proj_info
+    else:
+        return _get_projection_extension_metadata(gdal_info)
+
+
+def _get_projection_extension_metadata(gdal_info: dict) -> dict:
+    """Helper function that parses gdal.Info output without processing subdatasets.
+
+    :param gdal_info: Dictionary that contains the output from gdal.Info.
+
+    :return: Dictionary containing the info for the STAC extension for projections.
+        This dictionary contains the following fields, as described in stac-extensions,
+        see: https://github.com/stac-extensions/projection
+
+        - "proj:epsg"  The EPSG code of the CRS, if available.
+        - "proj:shape" The pixel size of the asset, if available.
+        - "proj:bbox"  The bounding box expressed in the asset CRS, if available.
+
+        When a field can not be found in the metadata that gdal.Info extracted,
+        we leave out that field.
+
+        Note that these dictionary keys in the return value *do* include the colon to be
+        in line with the names in stac-extensions.
+    """
+    proj_metadata = {}
+
+    # Size of the pixels
+    proj_metadata["proj:shape"] = gdal_info["size"]
+
+    # Extract the EPSG code from the WKT string
+    crs_as_wkt = gdal_info.get("coordinateSystem", {}).get("wkt")
+    if crs_as_wkt:
+        crs_id = extract_crs_epsg_code_from_wkt_string(crs_as_wkt)
+        if crs_id:
+            proj_metadata["proj:epsg"] = crs_id
+
+    # Size of the pixels
+    proj_metadata["proj:shape"] = gdal_info["size"]
+
+    # convert cornerCoordinates to proj:bbox format specified in
+    # https://github.com/stac-extensions/projection
+    # TODO: do we need to also handle 3D bboxes, i.e. the elevation bounds, if present?
+    if "cornerCoordinates" in gdal_info:
+        corner_coords = gdal_info.get("cornerCoordinates")
+        # TODO: check if this way to combine the corners also handles 3D bounding boxes correctly.
+        #   Need a correct example to test with.
+        lole = corner_coords["lowerLeft"]
+        upri = corner_coords["upperRight"]
+        proj_metadata["proj:bbox"] = [*lole, *upri]
+
+    return proj_metadata
+
+
+def extract_crs_epsg_code_from_wkt_string(crs_as_wkt: str) -> str:
```

Comment:
I would use CRS.from_wkt or CRS.from_user_input and then get epsg code out of it (CRS.to_epsg)

Reply:
Done.

Switched to using the CRS method from `pyproj.CRS` and removed `extract_crs_epsg_code_from_wkt_string`. 
This should indeed be more robust.